packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js

```diff
@@ -80,7 +85,24 @@ function SidebarNavigationScreenGlobalStylesContent() {
 			onChange={ noop }
 			onInput={ noop }
 		>
-			<StyleVariationsContainer />
+			<VStack
+				spacing={ 10 }
+				className="edit-site-global-styles-variation-container"
+			>
+				<StyleVariationsContainer />
+				<div>
+					<h3 className="edit-site-global-styles-variation-title">
+						{ __( 'Colors' ) }
+					</h3>
+					<ColorVariations />
+				</div>
+				<div>
+					<h3 className="edit-site-global-styles-variation-title">
+						{ __( 'Typography' ) }
+					</h3>
+					<TypographyVariations />
```

Comment:
I think we should check if variations exist before showing the headings. 

Here's emptytheme (no typography styles):

<img width="584" alt="Screenshot 2024-03-13 at 3 23 54 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/efa4319f-fcdc-448c-b01d-06a6a1f7d94c">

Reply:
Thanks, I updated the PR, though when I tested with emptytheme I don't see the above as I don't have any style variations :hmm:

Reply:
I ended up including some of the changes from https://github.com/WordPress/gutenberg/pull/59717 so that this works as expected

Reply:
Ah, I see now. It's failing the [uniqueTypographyVariations](https://github.com/WordPress/gutenberg/blob/e2982188dac463615b9e0ae826c625608ca608ab/packages/edit-site/src/components/global-styles/variations/variations-typography.js#L40-L40) test.

Empty theme has the following "typography" variation:

```
{
    "settings": {},
    "styles": {
        "elements": {
            "h2": {}
        },
        "blocks": {
            "core/post-title": {
                "typography": {
                    "fontWeight": "700"
                }
            }
        }
    },
    "title": "variation"
}
```

But it doesn't have any `fontFamilies`, so it fails. 

The same thing might happen for colors if there are no color settings:

<img width="335" alt="Screenshot 2024-03-14 at 3 11 51 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/bf34b752-658d-48ac-9c12-05cc9e9b3c4e">

I've pushed a change to check for this in a30eb5b. Seems to work okay, feel free to revert.